docs/tutorials/quantum_reinforcement_learning.ipynb

```diff
@@ -0,0 +1,4085 @@
+    "w_input, w_var, w_output = 1, 0, 2\n",
+    "if (('lambdas' not in model.trainable_variables[w_input].name)\n",
+    "    or ('thetas' not in model.trainable_variables[w_var].name)\n",
+    "    or ('obs-weights' not in model.trainable_variables[w_output].name)):\n",
+    "    raise ValueError(\"Wrong indexing of optimizers parameters.\")\n",
+    "    \n",
+    "def empty_circuits(n):\n",
+    "    return tfq.convert_to_tensor([cirq.Circuit()]*n)\n",
+    "\n",
+    "# Start training the agent\n",
+    "episode_reward_history = []\n",
+    "for batch in range(nb_batches):    \n",
+    "    loss_value = 0\n",
+    "    with tf.GradientTape() as tape:\n",
+    "        for episode in range(batch_size):\n",
+    "            action_probs_history = []\n",
+    "            rewards_history = []\n",
+    "            state = env.reset()\n",
+    "            episode_reward = 0\n",
+    "            done = False\n",
+    "            \n",
+    "            while not done:\n",
+    "                # Preprocess state\n",
+    "                state = tf.convert_to_tensor(state/state_bounds)\n",
+    "                state = tf.expand_dims(state, 0)\n",
+    "\n",
+    "                # Compute policy from env state\n",
+    "                action_probs = model([empty_circuits(1),state])\n",
+    "\n",
+    "                # Sample action from policy\n",
+    "                action = np.random.choice(nb_actions, p=action_probs[0].numpy())\n",
+    "                action_probs_history.append(tf.math.log(action_probs[0, action]))\n",
+    "\n",
+    "                # Apply sampled action in the environment, receive reward and next state\n",
+    "                state, reward, done, _ = env.step(action)\n",
+    "                rewards_history.append(reward)\n",
+    "                episode_reward += reward\n",
+    "            \n",
+    "            # Compute returns \n",
+    "            returns = []\n",
+    "            discounted_sum = 0\n",
+    "            for r in rewards_history[::-1]:\n",
+    "                discounted_sum = r + gamma * discounted_sum\n",
+    "                returns.insert(0, discounted_sum)\n",
+    "            # Normalize them for faster and more stable learning\n",
+    "            returns = np.array(returns)\n",
+    "            returns = (returns - np.mean(returns)) / (np.std(returns) + 1e-8)\n",
+    "            returns = returns.tolist()\n",
+    "            \n",
+    "            # Add contributions to batch loss\n",
+    "            history = zip(action_probs_history, returns)\n",
+    "            losses = []\n",
+    "            for log_prob, ret in history:\n",
+    "                losses.append(-log_prob * ret)\n",
+    "            loss_value += sum(losses)\n",
+    "            episode_reward_history.append(episode_reward)\n",
+    "            print(\"Episode {}/{}, rewards {}\".format(batch*batch_size+episode+1, batch_size*nb_batches, episode_reward))\n",
+    "        \n",
```

Comment:
I really don't see what "batch collecting trajectories" would do different? At the end of the day, the interaction with the environment is always gonna require evaluating the model at each step of the trajectory/episode, and this can only be done with batch_size 1.